thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/alert/content/EntityGroupByContentFormatter.java

```diff
@@ -0,0 +1,140 @@
+        LOG.warn("Anomaly result {} isn't an instance of MergedAnomalyResultDTO. Skip from alert.", anomalyResult);
+        continue;
+      }
+
+      MergedAnomalyResultDTO anomaly = (MergedAnomalyResultDTO) anomalyResult;
+      if (config == null) {
+        config = this.configDAO.findById(anomaly.getDetectionConfigId());
+        Preconditions.checkNotNull(config, String.format("Cannot find detection config %d", anomaly.getDetectionConfigId()));
+      }
+
+      updateEntityToAnomalyDetailsMap(anomaly, config);
+    }
+
+    List<Map.Entry<String, Double>> anomaliesSortedByScores = new ArrayList<>(entityAnomalyToScoreMap.entrySet());
+    anomaliesSortedByScores.sort(Map.Entry.comparingByValue());
+    List<String> entityList = anomaliesSortedByScores.stream().map(Map.Entry::getKey).collect(Collectors.toList());
+    entityList.sort(Collections.reverseOrder());
+
+    templateData.put("emailHeading", config.getName());
+    templateData.put("entityToAnomalyIdsMap", entityToAnomalyIdsMap);
+    templateData.put("entitySortedByScoreList", entityList);
+    templateData.put("entityToAnomalyDetailsMap", entityAnomalyReports.asMap());
+  }
+
+  /**
+   * Recursively find the anomalies having a groupKey and display them in the email
+   */
+  private void updateEntityToAnomalyDetailsMap(MergedAnomalyResultDTO anomaly, DetectionConfigDTO detectionConfig) {
+    if (anomaly.getProperties() != null && anomaly.getProperties().containsKey(PROP_GROUP_KEY) && anomaly.getChildIds() != null) {
+      AnomalyReportEntity anomalyReport = new AnomalyReportEntity(String.valueOf(anomaly.getId()),
+          getAnomalyURL(anomaly, emailContentFormatterConfiguration.getDashboardHost()),
+          ThirdEyeUtils.getRoundedValue(anomaly.getAvgBaselineVal()),
+          ThirdEyeUtils.getRoundedValue(anomaly.getAvgCurrentVal()), 0d, null,
+          getTimeDiffInHours(anomaly.getStartTime(), anomaly.getEndTime()), getFeedbackValue(anomaly.getFeedback()),
+          detectionConfig.getName(), detectionConfig.getDescription(), anomaly.getMetric(),
+          getDateString(anomaly.getStartTime(), dateTimeZone), getDateString(anomaly.getEndTime(), dateTimeZone),
+          getTimezoneString(dateTimeZone), getIssueType(anomaly));
+
+      // Criticality score ranges from [0 to infinity)
+      double score = -1;
+      if (anomaly.getProperties().containsKey(PROP_ANOMALY_SCORE)) {
+        score = Double.parseDouble(anomaly.getProperties().get(PROP_ANOMALY_SCORE));
+        anomalyReport.setScore(ThirdEyeUtils.getRoundedValue(score));
+      } else {
+        anomalyReport.setScore("-");
+      }
+      anomalyReport.setWeight(anomaly.getWeight());
+      anomalyReport.setGroupKey(anomaly.getProperties().get(PROP_GROUP_KEY));
+      anomalyReport.setEntityName(anomaly.getProperties().getOrDefault(PROP_ENTITY_NAME, "UNKNOWN_ENTITY"));
+
+      // include notified alerts only in the email
+      if (!includeSentAnomaliesOnly || anomaly.isNotified()) {
```

Comment:
Is the notified flag still in use?

Reply:
I am not 100% sure. We still have the legacy alerter running right.

Reply:
Right. We can keep it as a safety check.